### PR TITLE
Do not produce different files for each run on the mozilla-central repository

### PR DESCRIPTION
**Pull request by Luni-4**
When `rust-code-analysis-cli` was being run on the `mozilla-central` repository many times, the filenames of the output formats produced at each run were different. That was because the old code assigned filenames without considering that the thread scheduler picks up input files every time in a different order.

So, to avoid conflicts among files, we now use input file paths as output format filenames.

Thanks in advance for your review! :)

### CODE REVIEW

File: rust-code-analysis-cli/src/formats.rs [hunk 1]
```diff
@@ -62,22 +62,20 @@ impl Format {
...
 
-            let output_path = output_path.as_ref().unwrap();
+            // Use file path as filename
+            let filename = path.to_str().unwrap();
 
-            let mut file = path.as_path().file_name().unwrap().to_os_string();
-            file.push(format_ext);
+            // Remove ./ ../ .\ ..\ present in a path
+            let re = Regex::new(r"\.\./|\.\.\\|^\./|^\.\\").unwrap();
+            let no_dots_filename = re.replace_all(&filename, "");
```
Comment (marco-c) [inline rust-code-analysis-cli/src/formats.rs:70]: Is there some std lib function to do something like this?

File: rust-code-analysis-cli/src/formats.rs [hunk 2]
```diff
@@ -62,22 +62,20 @@ impl Format {
...
 
-            if format_path.as_path().exists() {
-                let mut new_filename = path.to_str().unwrap().to_string();
-                let re = Regex::new(r"[\\:/]").unwrap();
-                new_filename = re.replace_all(&new_filename, "_").to_string();
-                new_filename.push_str(format_ext);
-                format_path.pop();
-                format_path.push(new_filename);
-            }
+            // Create a clean filename
```
Comment (marco-c) [inline rust-code-analysis-cli/src/formats.rs:75]:
```suggestion
            // Create a clean filename by replacing `:` and slashes with `_`
```

File: rust-code-analysis-cli/src/formats.rs (hunk 1 above)
Comment (Luni-4) [reply rust-code-analysis-cli/src/formats.rs:70]: I think we could obtain the same result, reducing the code at the same time, using the [components](https://doc.rust-lang.org/std/path/struct.Path.html#method.components) function. I have to try though

**Review by marco-c — APPROVED** [review]
Looks much nicer now

**Pull request merged**